File: namer_app/lib/models/report.dart
enum ReportCategory { found, lost }

class Report {
  final String title;
  final String description;
  final ReportCategory category;
  final String location;
  final DateTime date;
  final String? imageUrl;

  Report({
    required this.title,
    required this.description,
    required this.category,
    required this.location,
    required this.date,
    this.imageUrl,
  });
}

File: loss_utal_backend/models/Report.js
const mongoose = require('mongoose');

const ReportSchema = new mongoose.Schema({
  title: { 
    type: String, 
    required: true 
  },
  description: { 
    type: String, 
    required: true 
  },
  category: { 
    type: String, 
    enum: ['found', 'lost'], 
    required: true 
  },
  location: { 
    type: String, 
    required: true 
  },
  latitude: { 
    type: Number 
  },
  longitude: { 
    type: Number 
  },
  imageUrl: { 
    type: String 
  },
  audioUrl: { 
    type: String 
  },
  userId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: 'User', 
    required: true 
  },
  status: { 
    type: String, 
    enum: ['active', 'resolved'], 
    default: 'active' 
  },
  createdAt: { 
    type: Date, 
    default: Date.now 
  }
});

module.exports = mongoose.model('Report', ReportSchema);
